cardano_clusterlib/clusterlib.py

```diff
@@ -1692,6 +1692,46 @@ def get_kes_period(self) -> int:
         """Return last block KES period."""
         return int(self.get_slot_no() // self.slots_per_kes_period)
 
+    def get_kes_period_info(
+        self, issue_op_cert: FileType, operational_certificate_issue_counter: FileType
+    ) -> Dict:
+        """Check an operational certificate and certificate issue counter file.
+
+        Args:
+            issue_op_cert: Operational certificate.
+            operational_certificate_issue_counter: Operational certificate counter.
+
+        Returns:
+            Dict: A dictionary containing all the information.
+        """
+        command_output = self.cli(
+            [
+                "query",
+                "kes-period-info",
+                *self.magic_args,
+                "--op-cert-file",
+                str(issue_op_cert),
+                "--operational-certificate-issue-counter-file",
+                str(operational_certificate_issue_counter),
+            ]
+        ).stdout
+
+        stdout_dec = command_output.decode("utf-8") if command_output else ""
+
+        # get output messages
+        messages_str = stdout_dec.split("{")[0]
+
+        if messages_str:
+            messages_list = list(filter(None, messages_str.split("\n")))
+
+        # get output metrics
+        metrics_str = stdout_dec.split("{")[-1]
+
+        if metrics_str and metrics_str.strip().endswith("}"):
+            metrics_dict = json.loads(f"{{{metrics_str}")
```

Comment:
If the `if` statement above evaluates as False, `metrics_dict` is not defined in the return statement below.